crates/astria-sequencer/src/grpc/sequencer.rs

```diff
@@ -152,6 +160,53 @@ impl SequencerService for SequencerServer {
 
         Ok(Response::new(block))
     }
+
+    #[instrument(skip_all)]
+    async fn get_pending_nonce(
+        self: Arc<Self>,
+        request: Request<GetPendingNonceRequest>,
+    ) -> Result<Response<GetPendingNonceResponse>, Status> {
+        use astria_core::primitive::v1::Address;
+
+        use crate::accounts::state_ext::StateReadExt as _;
+
+        let request = request.into_inner();
+        let Some(address) = request.address else {
+            trace!("required field address was not set",);
+            return Err(Status::invalid_argument(
+                "required field address was not set",
+            ));
+        };
+
+        let address = Address::try_from_raw(&address).map_err(|e| {
+            trace!(
+                error = %e,
+                "failed to parse address from request",
+            );
+            Status::invalid_argument(format!("invalid address: {e}"))
+        })?;
+        let nonce = self.mempool.pending_nonce(&address).await;
+
+        if let Some(nonce) = nonce {
+            return Ok(Response::new(GetPendingNonceResponse {
+                inner: nonce,
+            }));
+        }
+
+        // nonce wasn't in mempool, so just look it up from storage
+        let snapshot = self.storage.latest_snapshot();
+        let nonce = snapshot.get_account_nonce(address).await.map_err(|e| {
+            trace!(
```

Comment:
This strikes me as a critical error that we definitely want to notify on.
```suggestion
            error!(
```